lib/icinga/apiactions.cpp

```diff
@@ -555,5 +555,163 @@ Dictionary::Ptr ApiActions::GenerateTicket(const ConfigObject::Ptr&,
+		if (!EventCommand::GetByName(resolved_command))
+			return ApiActions::CreateResult(400, "Command '" + resolved_command + "' is not of type '" + command_type + "'.");
+	}
+
+	/* Get TTL param */
+	if (!params->Contains("ttl"))
+		return ApiActions::CreateResult(400, "ttl is required");
+
+	double ttl = HttpUtility::GetLastParameter(params, "ttl");
+	if (ttl <= 0)
+		return ApiActions::CreateResult(400, "ttl must be greater than 0");
+
+	/* This generates a UUID */
+	String uuid = Utility::NewUniqueID();
+
+	/* Create the deadline */
+	double deadline = Utility::GetTime() + ttl;
+
+	/* Update executions */
+	Dictionary::Ptr pending_execution = new Dictionary();
+	pending_execution->Set("pending", true);
+	pending_execution->Set("deadline", deadline);
+	Dictionary::Ptr executions = checkable->GetExecutions();
+	if (!executions)
+		executions = new Dictionary();
+	executions->Set(uuid, pending_execution);
+	checkable->SetExecutions(executions);
+
+	Endpoint::Ptr endpointPtr = Endpoint::GetByName(resolved_endpoint);
+	bool local = !endpointPtr || endpointPtr == Endpoint::GetLocalEndpoint();
+	if (local) {
+		/* TODO  */
+	} else {
+		/* Broadcast the update */
+		ApiListener::Ptr listener = ApiListener::GetInstance();
+		if (!listener)
+			return ApiActions::CreateResult(404, "No ApiListener instance available.");
+
+		Dictionary::Ptr updateParams = new Dictionary();
+		updateParams->Set("host", host->GetName());
+		if (service)
+			updateParams->Set("service", service->GetShortName());
+		updateParams->Set("executions", executions);
+
+		Dictionary::Ptr updateMessage = new Dictionary();
+		updateMessage->Set("jsonrpc", "2.0");
+		updateMessage->Set("method", "event::UpdateExecutions");
+		updateMessage->Set("params", updateParams);
+
+		MessageOrigin::Ptr origin = new MessageOrigin();
+		listener->SyncSendMessage(endpointPtr, updateMessage);
+
+		/* Execute command */
+		Dictionary::Ptr execMessage = new Dictionary();
+		execMessage->Set("jsonrpc", "2.0");
+		execMessage->Set("method", "event::ExecuteCommand");
+
+		/* TODO set the right params */
+		Dictionary::Ptr execParams = new Dictionary();
+		execMessage->Set("params", execParams);
+		execParams->Set("command_type", command_type);
+		execParams->Set("command", resolved_command);
+		execParams->Set("host", host->GetName());
+		if (service)
+			execParams->Set("service", service->GetShortName());
+
+		/*
+		 * If the host/service object specifies the 'check_timeout' attribute,
+		 * forward this to the remote endpoint to limit the command execution time.
+		 */
+		if (!checkable->GetCheckTimeout().IsEmpty())
+			execParams->Set("check_timeout", checkable->GetCheckTimeout());
+
+		execParams->Set("source", uuid);
+		execParams->Set("deadline", deadline);
+
+		listener->SyncSendMessage(endpointPtr, execMessage);
+	}
+
+	Dictionary::Ptr result = new Dictionary();
+	result->Set(checkable->GetName(), uuid);
+	return ApiActions::CreateResult(202, "Accepted", result);
+}
```

Comment:
Find this comment of mine in the files changed tab, not in the conversation one. Do you see the round red sign?